monai/apps/deepgrow/dataset.py

```diff
@@ -0,0 +1,271 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import logging
+import os
+import sys
+from typing import Callable, Dict, List, Sequence, Union
+
+import numpy as np
+
+from monai.apps.datasets import DecathlonDataset
+from monai.transforms import AsChannelFirstd, Compose, LoadImaged, Orientationd, Spacingd
+from monai.utils import GridSampleMode
+
+
+# TODO:: Test basic functionality
+# TODO:: Unit Test
+class DeepgrowDataset(DecathlonDataset):
+    def __init__(
+        self,
+        dimension: int,
+        pixdim: Sequence[float],
+        root_dir: str,
+        task: str,
+        section: str,
+        transform: Union[Sequence[Callable], Callable] = (),
+        download: bool = False,
+        seed: int = 0,
+        val_frac: float = 0.2,
+        cache_num: int = sys.maxsize,
+        cache_rate: float = 1.0,
+        num_workers: int = 0,
+        limit: int = 0,
+    ) -> None:
+        self.dimension = dimension
+        self.pixdim = pixdim
+        self.limit = limit
+
+        super().__init__(
+            root_dir=root_dir,
+            task=task,
+            section=section,
+            transform=transform,
+            download=download,
+            seed=seed,
+            val_frac=val_frac,
+            cache_num=cache_num,
+            cache_rate=cache_rate,
+            num_workers=num_workers,
+        )
+
+    def _generate_data_list(self, dataset_dir: str) -> List[Dict]:
+        dataset = super()._generate_data_list(dataset_dir)
+
+        tmp_dataset_dir = dataset_dir + "_{}.deep".format(self.section)
+        new_datalist = create_dataset(
+            datalist=dataset,
+            keys=["image", "label"],
+            output_dir=tmp_dataset_dir,
+            dimension=self.dimension,
+            pixdim=self.pixdim,
+            limit=self.limit,
+            relative_path=False,
+        )
+
+        dataset_json = os.path.join(tmp_dataset_dir, "dataset.json")
+        with open(dataset_json, "w") as fp:
+            json.dump({self.section: new_datalist}, fp, indent=2)
+        return new_datalist
+
+
+def _get_transforms(keys, pixdim):
+    mode = [GridSampleMode.BILINEAR, GridSampleMode.NEAREST] if len(keys) == 2 else [GridSampleMode.BILINEAR]
+    transforms = [
+        LoadImaged(keys=keys),
+        AsChannelFirstd(keys=keys),
+        Spacingd(keys=keys, pixdim=pixdim, mode=mode),
+        Orientationd(keys=keys, axcodes="RAS"),
+    ]
+
+    return Compose(transforms)
+
+
+def _save_data_2d(vol_idx, data, keys, dataset_dir, relative_path):
+    vol_image = data[keys[0]]
+    vol_label = data.get(keys[1])
+    data_list = []
+
+    if len(vol_image.shape) == 4:
+        logging.info("4D-Image, pick only first series; Image: {}; Label: {}".format(vol_image.shape, vol_label.shape))
```

Comment:
what will happen if the vol_label is None here?

Reply:
fixed